src/rgw/store/dbstore/common/dbstore.cc

```diff
@@ -1294,14 +1294,22 @@ int DB::Object::Read::read(int64_t ofs, int64_t end, bufferlist& bl, const DoutP
   bufferlist read_bl;
   uint64_t max_chunk_size = store->get_max_chunk_size();
 
+  // begin transaction to avoid object modification while reading
+  // chunks
+  int r = store->beginTransaction(dpp);
```

Comment:
i'm confused about how these transactions could be global to the store. if the goal is to support several rgw operations at once, don't each of the operations need a separate transaction, and the ability to associate their database ops with that transaction?

Reply:
These slides by Tom Lane seem to describe a system that works, in the context of a multi-threaded web+database application, the way we were elaborating in the prior discussion--there doesn't seem to be any mention of needing mutliple database handles matched to the different threads, but I might be wrong.  If that were the case, though, it shouldn't be necessary to keep a consistent thread handling the RGW operation--but rather, to associate the right handle with the operation, no (there'd be some other rules, too)?
https://www.postgresql.org/files/developer/concurrency.pdf